include/TPP/Dialect/Perf/PerfOps.td

```diff
@@ -0,0 +1,261 @@
+//===- TppOps.td - Tpp dialect ops -------------------------*- tablegen -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TPP_PERF_OPS
+#define TPP_PERF_OPS
+
+include "PerfDialect.td"
+include "mlir/Interfaces/ControlFlowInterfaces.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "mlir/IR/OpAsmInterface.td"
+
+def PerfTimer : AnyTypeOf<[I64]>;
+
+//===----------------------------------------------------------------------===//
+// BenchOp
+//===----------------------------------------------------------------------===//
+def Perf_BenchOp : Perf_Op<"bench",
+    [AutomaticAllocationScope, SingleBlockImplicitTerminator<"perf::YieldOp">]> {
+  let summary = "Benchmark the enclosed code.";
+  let description = [{
+    The `perf.bench` operation generates benchmarking code
+    around the enclosed region.
+    The performance results are gathered over the specified
+    number of iterations.
+
+    The operation allocates memory to store the results.
+    The lifetime of the returned results has to be managed by the user.
+
+    Example:
+
+    ```mlir
+
+    %deltas = perf.bench (%n) {
+      ... // ops under measurement
+    } -> memref<?xf64>
+
+    ```
+
+    Optionally, a yield op can be added:
```

Comment:
This comment is confusing what you want to say is mentioned in the yield op: If the parent operation defines no values, then the `perf.yield`...